static/js/chart/ranking_table.tsx

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for rendering a single ranking list.
+ */
+
+import React from "react";
+
+import { formatNumber, LocalizedLink } from "../i18n/i18n";
+import { formatString } from "../topic_page/string_utils";
+
+const NUM_FRACTION_DIGITS = 2;
+
+export interface Point {
+  dcid: string;
+  label: string;
+  value: number;
+  /**
+   * If not provided, the component will calculate the rank based on the order of the input points.
+   */
+  rank?: number;
+  /**
+   * Redirect link on the label. If not provided, the label will be in plain text.
+   */
+  redirectLink?: string;
+}
+
+interface RankingTablePropType {
+  title: string;
+  /**
+   * All the provided points will appear in the table as rows.
+   */
+  points: Point[];
+  /**
+   * If true, rows are ordered from highest ranks to lowest, e.g., 1,2,3...
+   * Otherwise, from lowest to highest, e.g., n,n-1,n-2,...
+   */
+  isHighest: boolean;
+  statVarName?: string;
+  unit?: string;
+  scaling?: number;
+  /**
+   * Total number of points. This is used for calculating ranks if isHighest is false
+   * (ordering from lowest ranks to highest) and ranks are not provided, e.g., n,n-1,n-2,...
+   */
+  numDataPoints?: number;
+  /**
+   * The row with dcid equal to currentDcid will be bolded.
+   */
+  currentDcid?: string;
+  /**
+   * If true, only the ranks and labels will be shown and not the values.
+   */
+  notShowValue?: boolean;
+}
+
+export function RankingTable(props: RankingTablePropType): JSX.Element {
+  // Calculates ranks based on the order of data if no rank is provided.
+  function getRank(
+    isHighest: boolean,
+    index: number,
+    numberOfTotalDataPoints?: number
+  ): number {
+    if (isHighest) {
+      return index + 1;
+    }
+    return numberOfTotalDataPoints
+      ? numberOfTotalDataPoints - index
+      : index + 1;
+  }
+
+  return (
+    <div className="ranking-table">
+      <h4>
+        {formatString(props.title, {
```

Comment:
why use formatString here? it looks like `props.title` is just going to be "Highest" or "Lowest" so there's nothing to replace in those titles?

Reply:
I tried to create a component that can be used by both the topic page and the place explorer page. This is the code copied from the [ranking unit component](https://github.com/datacommonsorg/website/blob/master/static/js/topic_page/ranking_unit.tsx#L45-L50). It will show the stats var in the title.

Reply:
Done